dissect/target/plugins/apps/av/mcafee.py

```diff
@@ -0,0 +1,119 @@
+
+    @export(record=McAfeeMscLogRecord)
+    def msc(self) -> Iterator[McAfeeMscLogRecord]:
+        """Return msc log history records from McAfee.
+
+        Yields McAfeeMscLogRecord with the following fields:
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            ts (datetime): timestamp.
+            ip (net.ipadress): IP of suspicious connection (if available).
+            tcp_port (net.tcp.Port): TCP Port of suspicious incoming connection (if available).
+            udp_port (net.udp.Port): UDP Port of suspicious incoming connection (if available).
+            threat (string): Description of the detected threat (if available).
+            message (string): Message as reported in the user interface (might include template slots).
+            keywords (string): Unparsed fields that might be visible in user interface.
+            fkey (string): Foreign key for reference for further investigation.
+        """
+
+        len_udpstr = len(self.MARKER_SUSPICIOUS_UDP_CONNECTION)
+        len_tcpstr = len(self.MARKER_SUSPICIOUS_TCP_CONNECTION)
+
+        for log_file in self.get_log_files():
+            with log_file.open() as open_log:
+                database = SQLite3(open_log)
+                fields = {}
+                fields_table = database.table(self.TABLE_FIELD)
+                for field in fields_table.rows():
+                    if not fields.get(field.fkey):
+                        fields[field.fkey] = {}
+                    fields[field.fkey][field.field_id] = field.data
+                log_table = database.table(self.TABLE_LOG)
+                for entry in log_table.rows():
+                    fkey = entry.fkey
+                    log_fields = fields.get(fkey, {})
+                    ip = None
+                    protocol = None
+                    port = None
+                    threat = None
+                    for key, log_field in log_fields.items():
+                        try:
+                            ipaddress.ip_address(log_field)
+                            ip = log_field
+                            continue
+                        except ValueError:
+                            pass
+                        if log_field.startswith(self.MARKER_SUSPICIOUS_TCP_CONNECTION):
+                            port = int(log_field[len_tcpstr:])
+                            protocol = "TCP"
+                            continue
+                        if log_field.startswith(self.MARKER_SUSPICIOUS_UDP_CONNECTION):
+                            port = int(log_field[len_udpstr:])
+                            protocol = "UDP"
+                            continue
+                        if key == self.TEMPLATE_ID_INFECTION and entry.details_info.find(self.MARKER_INFECTION) > -1:
+                            threat = log_field
+                    yield McAfeeMscLogRecord(
+                        ts=entry.date,
```

Comment:
Can you check if this has a timezone attached? If not, maybe best to fix this directly in `dissect.sql`?

Reply:
No timezone attached, stored in SQLite as a unix timestamp, so also no timezone needed. I think they just use the timezone of the host upon display.

![image](https://user-images.githubusercontent.com/110607403/225888231-9340c22a-0013-4d95-a27d-325cdb8aa303.png)

Reply:
Unix timestamps are always UTC, so a `tz=timezone.utc` needs to be added to create aware datetime objects. Also see https://docs.python.org/3/library/datetime.html#aware-and-naive-objects

Reply:
I see that flow.record converts this to utc if passed as int so that should give us a uniform datetime record even if the timezones on the hosts may vary, this is quite optimal right?

Reply:
`flow.record` does not currently create aware datetime objects, this is a known issue and being considered (it has a significant performance hit).

My question was if this returned a datetime object or an integer. I didn't remember if SQLite has a dedicated datetime type but it turns out it does not. So since this returns an integer, you must add a `ts.from_unix(entry.date)` to this line.

Reply:
Ah I see because the flow.record constructor uses utcfromtimestamp which leaves a naive dt object